src/packageurl/contrib/url2purl.py

```diff
@@ -431,54 +455,54 @@ def build_github_api_purl(url):
 # https://codeload.github.com/nexB/scancode-toolkit/tar.gz/v3.1.1
 # https://codeload.github.com/berngp/grails-rest/zip/release/0.7
 github_codeload_pattern = (
-    r"https?://codeload.github.com/(?P<namespace>.+)/(?P<name>.+)/"
-    r"(zip|tar.gz|tar.bz2|tgz)/(.*/)*"
+    r"https?://codeload\.github\.com/(?P<namespace>.+)/(?P<name>.+)/"
+    r"(zip|tar\.gz|tar\.bz2|tgz)/([^/]*/)*"
     r"(?P<version_prefix>v|V?)(?P<version>.+)$"
 )
 
 register_pattern("github", github_codeload_pattern)
 
 
-@purl_router.route("https?://github\\.com/.*")
+@purl_router.route(r"https?://github\.com/.*")
 def build_github_purl(url):
     """
     Return a PackageURL object from GitHub `url`.
     """
 
     # https://github.com/nexB/scancode-toolkit/archive/v3.1.1.zip
     archive_pattern = (
-        r"https?://github.com/(?P<namespace>.+)/(?P<name>.+)"
-        r"/archive/(.*/)*"
+        r"https?://github\.com/(?P<namespace>.+)/(?P<name>.+)"
+        r"/archive/([^/]*/)*"
         r"((?P=name)(-|_|@))?"
-        r"(?P<version_prefix>v|V?)(?P<version>.+).(zip|tar.gz|tar.bz2|.tgz)"
+        r"(?P<version_prefix>v|V?)(?P<version>.+)\.(zip|tar\.gz|tar\.bz2|tgz)"
     )
 
     # https://github.com/downloads/mozilla/rhino/rhino1_7R4.zip
     download_pattern = (
-        r"https?://github.com/downloads/(?P<namespace>.+)/(?P<name>.+)/"
+        r"https?://github\.com/downloads/(?P<namespace>.+)/(?P<name>.+)/"
         r"((?P=name)(-|@)?)?"
-        r"(?P<version_prefix>v|V?)(?P<version>.+).(zip|tar.gz|tar.bz2|.tgz)"
+        r"(?P<version_prefix>v|V?)(?P<version>.+)\.(zip|tar\.gz|tar\.bz2|tgz)"
```

Comment:
This also fixes a bug with tgz files, since it would previously match

`v1.2.3.tgz` as `{version: "1.2.", extension: ".tgz"}`
as opposed to `{version: "1.2.3", extension: "tgz"}`

due to the double unescaped `.` here